dashboard/app/models/user.rb

```diff
@@ -1043,7 +1044,7 @@ def upgrade_to_teacher(email, email_preference = nil)
 
       # Remove family name, in case it was set on the student account.
       if DCDO.get('family-name-features', false)
-        properties['family_name'] = nil
+        self.family_name = nil
```

Comment:
out of curiousity, is the `self` bit needed here?

Reply:
seems like it, unfortunately. I tried it without first and it seemed like it was just making a new variable, and the tests failed. if there's a cleaner way to do this i'm all for it though